### PR TITLE
Add Catch2 string conversions for `result<T, E>` and `error`.

### PR DESCRIPTION
**Pull request by de-vri-es**
This should help a lot with unit tests. Copied it from `fizyr_rpc` basically.

Sadly, we can not include `catch.hpp` from the header, because we don't know how to find the header. However, it does give a clear error message if it is included before `catch.hpp`.

### CODE REVIEW

File: include/estd/result/catch_string_conversions.hpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+#include "./result.hpp"
+#include "./error.hpp"
+
+// This header adds Catch2 string conversions for estd::result<T, E> and estd::error,
+// for pretty messages with the Catch2 unit testing framework.
+
+#if !defined(CATCH_VERSION_MAJOR) && !defined(TWOBLUECUBES_CATCH_HPP_INCLUDED)
+// You must already have included the Catch2 header, since we can not know how to include it properly.
+// Some projects embed `catch.hpp`, other use a system include.
```
Comment (vcarpani) [inline include/estd/result/catch_string_conversions.hpp:11]: I think we should force system include on all of our repos and not encourage bad including

Comment (de-vri-es) [reply include/estd/result/catch_string_conversions.hpp]: Fair enough, switched `estd` to system Catch2, and now including `<catch2/catch.hpp>`.

**Comment by de-vri-es** [comment]
Added comments and processed the other point. Waiting for the whole workspace tests to build before merging :)

**Comment by de-vri-es** [comment]
Tests build (wish I could say they also all passed).

**Pull request merged**